test/controllers/property.controller.test.js

```diff
@@ -1030,3 +1030,103 @@ describe('Get all properties added by an admin', () => {
     });
   });
 });
+
+describe('Search Through Properties', () => {
+  const id = mongoose.Types.ObjectId();
+  const property = PropertyFactory.build({
+    _id: id,
+    addedBy: _id,
+    updatedBy: _id,
+    houseType: '3 bedroom duplex',
+    state: 'lagos',
+    area: 'lekki',
+  });
+
+  const filter = {
+    houseType: '3 bedroom duplex',
+    state: 'lagos',
+    area: 'lekki',
+  };
+
+  context('when no property is found', () => {
+    it('returns not found', (done) => {
+      request()
+        .post('/api/v1/property/search')
+        .set('authorization', userToken)
+        .send(filter)
+        .end((err, res) => {
+          expect(res).to.have.status(200);
+          expect(res.body.success).to.be.eql(true);
+          expect(res.body.properties.length).to.be.eql(0);
+          done();
+        });
+    });
+  });
+
+  describe('when properties exist in db', () => {
+    beforeEach(async () => {
+      await addProperty(property);
+    });
+
+    context('with a valid token & id', () => {
+      it('returns successful payload', (done) => {
+        request()
+          .post('/api/v1/property/search')
+          .set('authorization', userToken)
+          .send(filter)
+          .end((err, res) => {
+            expect(res).to.have.status(200);
+            expect(res.body.success).to.be.eql(true);
+            expect(res.body).to.have.property('properties');
+            expect(property._id.equals(res.body.properties[0]._id)).to.be.eql(true);
```

Comment:
I have corrected this in other PRs, a test is more readable if it contains expectation in the expect functiona and the actual value in the comparison function.
```suggestion
            expect(res.body.properties[0]._id).to.be.eql(property._id);
```